lib/puppet/graph/relationship_graph.rb

```diff
@@ -244,6 +244,7 @@ def replace_containers_with_anchors(catalog)
     containers.each { |x|
       admissible[x] = whit_class.new(:name => "admissible_#{x.ref}", :catalog => catalog)
       completed[x]  = whit_class.new(:name => "completed_#{x.ref}",  :catalog => catalog)
+      admissible[x].set_tags(x)
```

Comment:
I think it's probably worth adding a code comment here describing why we're doing this, as well as that `set_tags` is explicitly going to copy the tags (as opposed to set them to a reference to the original resource's tags).